Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Animations;
+
+public class Checkpoint : MonoBehaviour {
+    [SerializeField]
+    public bool isActivated = false;
+   
+    public Animator anim;
```

Comment:
avoid public fields. Either use SerializeField if all you need to do is expose to editor, or use public Properties to expose variables to other classes. 
* https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties
* https://unity3d.com/learn/tutorials/topics/scripting/properties